### ISSUE TITLE
How to use the new Projected Coordinate System Reference-GUI

### ISSUE THREAD

**Issue by eibre** [Autodesk/revit-ifc#255]
Can someone guide me on how to fill inn the correct information in the new GUI?
* How to fill in Name, Description and Geodetic Datum? I tried to add the ProjectGlobalPositioning.CRSName shared parameter to ProjectInfo, but nothing shows in the form.
* Eastings and Northings: Are models exportet with rotation now, if Project Base Point is selected as Coordinate Base? Is thee a way to set these values independent of the shared coordinates?
* EPSG Code: When I input a number here, it is gone the next time I open the dialog.
![image](https://user-images.githubusercontent.com/5968613/101879813-2d86b900-3b92-11eb-93f3-0a95b94b4ae2.png)

**Comment by AngelVelezSosa** [comment]
@WawanSolihin .

**Comment by WawanSolihin** [comment]
The default value for the Projected Coordinate System Reference will be taken from Revit SiteLocation. It is typically imported from AutoCAD or Civil3D.
Alternatively, when such information is not available, you can enter the EPSG code. For example 25832 for ETRS89 / UTM zone 32N (from http://epsg.io), When the focus leaves the filed (by pressing Tab, or moving the cursor to another field), it will automatically search and fill in the other information based on the internal DB. The Eastings and Northings will be dependent on the Coordinate Base chosen.
You may refer to https://github.com/Autodesk/revit-ifc/wiki/Mapping-geo-reference-in-IFC4 for some explanation how it works.

**Comment by eibre** [comment]
I tried with 5950 http://epsg.io/5950 but it didn't work. Are not all EPSG-codes supported? 25832 + tab did work.

**Comment by eibre** [comment]
Why is it only "Shared Coordinates" that rotates the ifc according to true north? (when opened in Solibri) It seems that the eastings an nothings are measured along different axis when Project Base Point is chosen versus Internal Origin, shouldn't that be reflected by the orientation of the model in the ifc file?

**Comment by WawanSolihin** [comment]
There is no rotation in any sort. The illustrations show that on the ground the project looks like item 1 in the picture. In Revit it is usually placed in certain rotation convenient that cause the True North no longer pointing to the actual North (the project point +Y is) (picture 2). The rest describes how this can be used to correctly map so that Revit IFC exporter can get the right values for Eastings and Northings. The XAxisAbscisa and XAxisOrdinate is the direction of the X-axis in relation to the True North

**Comment by WawanSolihin** [comment]
> I tried with 5950 http://epsg.io/5950 but it didn't work. Are not all EPSG-codes supported? 25832 + tab did work.

It seems that not all codes you get from epsg.io is recognized. The data comes from AcDbGeoData::coordinateSystem. I think it takes the information from "C:\ProgramData\Autodesk\Geospatial Coordinate Systems 14.07\NameMapper.csv". If you look at the content it does not have 5950, but it has the larger one EPSG:4258

**Comment by eibre** [comment]
> `There is no rotation in any sort.` 
Yes, if you choose "Current Shared Coordinates" the IFC file will be rotated according to true north when opened in Solibri.

Is it possible to add the CRS-info manually if the EPSG code isn't recognized?

**Comment by WawanSolihin** [comment]
Currently not because we want to make sure the information is correct and consistent. Maybe we will provide a way to override in the next update, but it will be user responsibility to input the correct information

**Comment by ziyike** [comment]
@WawanSolihin hey, it's been more than a year since this discussion, it seems like this issue still persists.
While understanding the idea of making sure the information for IfcProjectedCRS being correct, not supported seems worse than being wrong, right?

It seems like a different approach is provided, by mannually entering parameters like ProjectGlobalPositioning.CRSName and more may allow us to export to a usable IFC file... yet not quite, the information regarding ProjectGlobalPositioning are not fully utilized, if at all.

So, any suggestions?
Please?

**Comment by ziyike** [comment]
@WawanSolihin Much thanks for the information! Guess we have to provide addins to support this in previous releases on our own, cuz many of our user groups are stuck at version 2020.

**Comment by ziyike** [comment]
@AngelVelezSosa Thanks for your response! But I came here for a solution, not a debate.

**Comment by AngelVelezSosa** [comment]
Hi, @ziyike , not trying to start a debate but trying to understand the requirement.  I often hear that an incorrect answer is worse than no answer, so wanted to hear if that was true for you or not.  Perhaps there are cased where something thar is approximately correct is better than no guidance.

**Comment by ArtemHnatko** [comment]
Hi @eibre, thank you for raising this — and to everyone who followed up in this thread.

We have reviewed the three concerns from this issue against the current codebase. Here is where things stand.

1. Name / Description / Geodetic Datum fields not populated from shared parameters

This behavior was intentional by the time of the Revit 2023 release. The ProjectGlobalPositioning.* shared parameters have been deprecated and are no longer read by the export dialog. CRS information is now entered directly in the Geographic Reference tab of the IFC Export dialog and is saved as part of the export configuration — so it persists across sessions as long as you are using a named (non-in-session) configuration.

2. Eastings / Northings axis direction differs by Coordinate Base; only Shared Coordinates rotates to True North

This is expected behavior. The True North rotation is encoded in IfcMapConversion.XAxisAbscissa / XAxisOrdinate and is only applied when Shared Coordinates is selected as the Coordinate Base, because that is the only mode in which Revit has a well-defined relationship between project north and true north. Other coordinate base modes (Project Base Point, Internal Origin) shift the origin but do not carry a true-north rotation. This limitation is tracked separately in [#177](https://github.com/Autodesk/revit-ifc/issues/177).

3. EPSG code disappears when the dialog is reopened

This depends on which EPSG code is used and which configuration type is active.

- Recognized codes (those present in C:\ProgramData\Autodesk\Geospatial Coordinate Systems xx.xx\NameMapper.csv) are correctly stored and restored for any saved named configuration.
- Unrecognized codes (e.g., EPSG 5950) cause the SetGeoCoordinateSystem call to fail silently; the dialog clears the field immediately on focus loss and the value is never written to the configuration. There is currently no way to force-save an unrecognized code through the UI.
- In-session configuration is never persisted to the document; all CRS values entered there are lost when the dialog is closed.

The underlying request for a proper EPSG registry picker (rather than a free-text field backed by a local CSV) is tracked in [#183](https://github.com/Autodesk/revit-ifc/issues/183) / Jira REVIT-238312 and remains open.

We are closing this issue as the three original concerns have been addressed:

- The shared-parameters question (# 1) is resolved — that workflow was deprecated and replaced with direct GUI input in Revit 2023.
- The coordinate base / orientation behavior (# 2) is by design and is tracked in the dedicated issue [#177](https://github.com/Autodesk/revit-ifc/issues/177).
- The EPSG persistence problem (# 3) is resolved for recognized codes in saved configurations. The remaining gap — support for codes not present in the Autodesk Geospatial database — is a separate feature request tracked in [#183](https://github.com/Autodesk/revit-ifc/issues/183) / Jira REVIT-238312.

If you encounter a regression or a scenario not covered by the workarounds above, please open a new issue with a minimal reproduction case and we will investigate.
